examples/service-clients/azure-client/external-controller/src/controller.ts

```diff
@@ -67,18 +72,22 @@ export class DiceRollerController
 		});
 	}
 
-	public get value(): number {
+	public get value(): DieValue {
 		const value = this.props.get(diceValueKey);
 		if (typeof value !== "number") {
 			throw new TypeError(
 				"Model is incorrect - did you call DiceRollerController.initializeModel() to set it up?",
 			);
 		}
-		return value;
+		if (value < 1 || value > 6) {
+			throw new RangeError("Model is incorrect - value is out of range");
+		}
+		return value as DieValue;
 	}
 
 	public readonly roll = (): void => {
-		const rollValue = Math.floor(Math.random() * 6) + 1;
+		const rollValue = (Math.floor(Math.random() * 6) + 1) as DieValue;
 		this.props.set(diceValueKey, rollValue);
+		this.onSet(rollValue);
```

Comment:
I may be misreading this, but since you are setting props on changes of your own values, does it mean that user has to handle the changes of own local values? In other words, they cannot listen to them like any other client values changes?

Reply:
`onSet` is a callback from the controller to user. Think super cheap and very specific event. I'll add some comments.